Source/ui_unix/mainwindow.cpp

```diff
@@ -0,0 +1,228 @@
+
+    connect(openglpanel, SIGNAL(keyUp(QKeyEvent*)), this, SLOT(keyReleaseEvent(QKeyEvent*)));
+    connect(openglpanel, SIGNAL(keyDown(QKeyEvent*)), this, SLOT(keyPressEvent(QKeyEvent*)));
+}
+
+MainWindow::~MainWindow()
+{
+    delete ui;
+}
+
+void MainWindow::showEvent(QShowEvent* event)
+{
+    QMainWindow::showEvent(event);
+    initEmu();
+}
+
+void MainWindow::initEmu(){
+    g_virtualMachine = new CPS2VM();
+    g_virtualMachine->Initialize();
+
+    g_virtualMachine->CreateGSHandler(CGSH_OpenGLQt::GetFactoryFunction(openglpanel));
+    SetupSoundHandler();
+
+    g_virtualMachine->CreatePadHandler(CPH_HidUnix::GetFactoryFunction());
+    padhandler = (CPH_HidUnix*)g_virtualMachine->GetPadHandler();
+
+    StatsManager = new CStatsManager();
+    g_virtualMachine->m_ee->m_gs->OnNewFrame.connect(boost::bind(&CStatsManager::OnNewFrame, StatsManager, _1));
+
+    createStatusBar();
+
+    fpstimer = new QTimer(this);
+    connect(fpstimer, SIGNAL(timeout()), this, SLOT(setFPS()));
+
+    Setupfpscounter();
+}
+
+void MainWindow::setOpenGlPanelSize()
+{
+    openGLWindow_resized();
+}
+
+void MainWindow::SetupSoundHandler()
+{
+    if(g_virtualMachine != nullptr){
+        bool audioEnabled = CAppConfig::GetInstance().GetPreferenceBoolean(PREFERENCE_AUDIO_ENABLEOUTPUT);
+        if(audioEnabled)
+        {
+            g_virtualMachine->CreateSoundHandler(&CSH_OpenAL::HandlerFactory);
+        }
+        else
+        {
+            g_virtualMachine->DestroySoundHandler();
+        }
+    }
+}
+
+void MainWindow::Setupfpscounter()
+{
+    if(g_virtualMachine != nullptr){
+        bool fpsEnabled = CAppConfig::GetInstance().GetPreferenceBoolean(PREFERENCE_UI_SHOWFPS);
+        if(fpsEnabled)
+        {
+            fpsLabel->show();
+            dcLabel->show();
+            fpstimer->stop();
+            fpstimer->start(1000);
+        }
+        else
+        {
+            fpstimer->stop();
+            fpsLabel->hide();
+            dcLabel->hide();
+        }
+    }
+}
+
+void MainWindow::openGLWindow_resized()
+{
+    if (g_virtualMachine != nullptr && g_virtualMachine->m_ee != nullptr  && g_virtualMachine->m_ee->m_gs != nullptr )
+        {
+
+            GLint w = openglpanel->size().width(), h = openglpanel->size().height();
+
+            CGSHandler::PRESENTATION_PARAMS presentationParams;
+            presentationParams.mode 			= (CGSHandler::PRESENTATION_MODE)CAppConfig::GetInstance().GetPreferenceInteger(PREF_CGSHANDLER_PRESENTATION_MODE);
+            presentationParams.windowWidth 		= w;
+            presentationParams.windowHeight 	= h;
+            g_virtualMachine->m_ee->m_gs->SetPresentationParams(presentationParams);
+            g_virtualMachine->m_ee->m_gs->Flip();
+        }
+}
+
+void MainWindow::on_actionOpen_Game_triggered()
+{
+    QFileDialog dialog(this);
+    dialog.setFileMode(QFileDialog::ExistingFile);
+    dialog.setNameFilter(tr("ISO/CSO/ISZ files (*.iso *cso *.isz)"));
+    QString fileName;
```

Comment:
`fileName` should be declared inside the if's scope since it's not used anywhere outside. Should also use `auto` type deduction like this:
`auto fileName = dialog.selectedFiles().first();`